fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationSettings.kt

```diff
@@ -0,0 +1,148 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Divider
+import org.mozilla.fenix.compose.SwitchWithLabel
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.list.TextListItem
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Firefox Translation settings fragment compose view.
+ */
+@Composable
+fun TranslationSettings(
+    translationSwitchList: List<TranslationSwitchItem>,
+    onAutomaticTranslationClicked: () -> Unit,
+    onNeverTranslationClicked: () -> Unit,
+    onDownloadLanguageClicked: () -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .background(
+                color = FirefoxTheme.colors.layer1,
+            ),
+    ) {
+        LazyColumn {
+            items(translationSwitchList) { item: TranslationSwitchItem ->
+                SwitchWithLabel(
+                    checked = item.isEnabled,
+                    onCheckedChange = item.onStateChange,
+                    label = item.textLabel,
+                    modifier = Modifier
+                        .padding(start = 72.dp, end = 16.dp)
+                        .clearAndSetSemantics {
+                            contentDescription = item.contentDescription
+                        },
+                )
+
+                if (item.hasDivider) {
+                    Divider(Modifier.padding(top = 8.dp, bottom = 8.dp))
+                }
+            }
+        }
+
+        Text(
+            text = stringResource(
+                id = R.string.translation_settings_fragment_translation_preference,
+            ),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 72.dp, end = 16.dp, bottom = 16.dp, top = 8.dp),
+            color = FirefoxTheme.colors.textAccent,
+            style = FirefoxTheme.typography.headline8,
+        )
+
+        TextListItem(
+            label = stringResource(id = R.string.translation_settings_fragment_automatic_translation),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 56.dp)
+                .clickable { onAutomaticTranslationClicked() },
```

Comment:
For the `clickable`, could we move this into using the API's `onClick` https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt#55

Reply:
Done